### DISCUSSION TITLE
Cannot generate two pydantic models from the same SQLAlchemy model

### DISCUSSION THREAD

**Discussion by saschahofmann** [tiangolo/pydantic-sqlalchemy#88] [Questions] [answered]
For the create method of SQL model I need to exclude the id of a model, I'd like to call the `sqlalchemy_to_pydantic` function twice on the same model for example like this:

```python
PyUser = sqlalchemy_to_pydantic(User)
PyUserCreate = sqlalchemy_to_pydantic(User, exclude=['id'])
```

Now the server and every thing starts just fine but when I try to fetch the *openapi.json*, I get a keyerror because the first model is not in the model map. 

I think line 36 in *main.py* is at fault
```python
pydantic_model = create_model(
        db_model.__name__, __config__=config, **fields  # type: ignore
    )
```
The `db_model.__name__` is the same two times so it must be overwriting some key (Disclaimer: I have no idea about the inner workings of pydantic but it seems to be creating some kind of global map of all models?)

I suggest to add an optional parameter `name` to the function and replace that line.

**Comment by tiangolo** [comment, marked as answer]
Thanks for the patience with my reply! :sweat_smile:

I deprecated this library a while ago in favor of SQLModel which would solve all the same use cases in a much better way, and I'm directing all the effort in that direction.

SQLModel uses Pydantic models to define both data and table (SQL) models, so it's easier to do all these things, including that, it would be much, much simpler to do. :rocket:
